src/api/objects/ObjectAPI.js

```diff
@@ -501,7 +501,7 @@ export default class ObjectAPI {
      * @method mutate
      * @memberof module:openmct.ObjectAPI#
      */
-    mutate(domainObject, path, value) {
+    async mutate(domainObject, path, value) {
```

Comment:
Right now "mutation" refers to the mutation of a domain object in-memory, and the fact that it also saves the object in some cases is a side effect (an unexpected one I would argue, I've never been very happy with this).

So, right now the mutation itself is synchronous, and I think our interface should reflect that...for now.

I think we will move to a mental model where we consider mutation to be of a _distributed_ object which is held by many simultaneous users, in which case, mutation will be asynchronous, but we're not quite there yet.

Making `mutate` asynchronous now fundamentally changes our interface contract and may break code that is depending on it to be synchronous.

Reply:
Ok, yeah, it was throwing me off with the asynchronous 'save' in there. But that makes sense, thank you for that explanation, this mix of mutate and save has been bugging me for a while.